distributed/shuffle/tests/test_shuffle_extension.py

```diff
@@ -115,3 +120,231 @@ def test_split_by_partition():
     assert set(out) == {1, 2, 3}
     assert out[1].column_names == list(df.columns)
     assert sum(map(len, out.values())) == len(df)
+
+
+def test_rechunk_slicing_1():
```

Comment:
I would likely rather put these test into the rechunk test module. Similar change for the actual functionality.